examples/md-flexible/src/Simulation.cpp

```diff
@@ -21,346 +22,235 @@ extern template bool autopas::AutoPas<ParticleType>::iteratePairwise(autopas::LJ
   }
 
-  _timers.init.stop();
-}
-
-template <class FunctorType>
-void Simulation::calculateForces(autopas::AutoPas<ParticleType> &autopas) {
-  _timers.forceUpdateTotal.start();
-
-  // pairwise forces
-
-  _timers.forceUpdatePairwise.start();
+  _configuration.flushParticles();
 
-  FunctorType functor{autopas.getCutoff(), *_particlePropertiesLibrary};
-  bool tuningIteration = autopas.iteratePairwise(&functor);
-
-  _timers.forceUpdateTotal.stop();
-  auto timeIteration = _timers.forceUpdatePairwise.stop();
-
-  // count time spent for tuning
-  if (tuningIteration) {
-    _timers.forceUpdateTuning.addTime(timeIteration);
-    ++_numTuningIterations;
-  } else {
-    _timers.forceUpdateNonTuning.addTime(timeIteration);
-    // if the previous iteration was a tuning iteration an the current one is not
-    // we have reached the end of a tuning phase
-    if (_previousIterationWasTuningIteration) {
-      ++_numTuningPhasesCompleted;
+  if (_configuration.useThermostat.value and _configuration.deltaT.value != 0) {
+    if (_configuration.addBrownianMotion.value) {
+      Thermostat::addBrownianMotion(*_autoPasContainer, *(_configuration.getParticlePropertiesLibrary()),
+                                    _configuration.initTemperature.value);
     }
+    Thermostat::apply(*_autoPasContainer, *(_configuration.getParticlePropertiesLibrary()),
+                      _configuration.initTemperature.value, std::numeric_limits<double>::max());
   }
-  _previousIterationWasTuningIteration = tuningIteration;
-
-  // global forces
 
-  _timers.forceUpdateTotal.start();
-  _timers.forceUpdateGlobal.start();
-  globalForces(autopas);
-  _timers.forceUpdateGlobal.stop();
-  _timers.forceUpdateTotal.stop();
+  _timers.initialization.stop();
 }
 
-void Simulation::globalForces(autopas::AutoPas<ParticleType> &autopas) {
-  // skip application of zero force
-  if (_config->globalForceIsZero()) {
-    return;
+Simulation::~Simulation() { _timers.total.stop(); }
+
+void Simulation::run() {
+  const int iterationsPerSuperstep = _configuration.verletRebuildFrequency.value;
+  _timers.simulate.start();
+  for (int i = 0; i < _configuration.iterations.value; i += iterationsPerSuperstep) {
+    executeSupersteps(iterationsPerSuperstep);
   }
+  _timers.simulate.stop();
 
-#ifdef AUTOPAS_OPENMP
-#pragma omp parallel default(none) shared(autopas)
-#endif
-  for (auto particleItr = autopas.begin(autopas::IteratorBehavior::owned); particleItr.isValid(); ++particleItr) {
-    particleItr->addF(_config->globalForce.value);
+  // Record last state of simulation.
+  if (_createVtkFiles) {
+    _vtkWriter->recordTimestep(_iteration, *_autoPasContainer);
   }
 }
 
-void Simulation::simulate(autopas::AutoPas<ParticleType> &autopas) {
-  this->_homogeneity = Simulation::calculateHomogeneity(autopas);
-  _timers.simulate.start();
+void Simulation::executeSupersteps(const int iterationsPerSuperstep) {
+  for (int i = 0; i < iterationsPerSuperstep && needsMoreIterations(); ++i) {
```

Comment:
```suggestion
  for (int i = 0; i < iterationsPerSuperstep and needsMoreIterations(); ++i) {
```
consistency with the rest of the project